src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoader.cs

```diff
@@ -355,6 +355,17 @@ private static void LoadAllocations(ChainSpecJson chainSpecJson, ChainSpec chain
             return;
         }
 
+        Dictionary<Hash256, byte[]> codes = new();
```

Comment:
I think we don't need this dictionary and can just use `chainSpecJson.CodeHashes` directly later to conserve a bit of memory.